FSharp.AutoComplete/Program.fs

```diff
@@ -607,6 +631,64 @@ module internal Main =
 
             main state
 
+
+          | Methods ->
+            // Find the starting point, ideally right after the first '('
+            let line, col =
+              let rec prevPos (line,col) =
+                match line, col with
+                | 1, 1 -> 1, 1
+                | _, 1 ->
+                   let prevLine = state.Files.[file].Lines.[line - 2]
+                   if prevLine.Length = 0 then prevPos(line-1, 1)
+                   else line - 1, prevLine.Length
+                | _    -> line, col - 1
+
+              let rec loop depth (line, col) =
+                if (line,col) <= (1,1) then (line, col) else
+                let ch = state.Files.[file].Lines.[line - 1].[col - 1]
+                if (ch = '(' || ch = '{' || ch = '[') && depth > 0 then loop (depth - 1) (prevPos (line,col))
```

Comment:
This returns the character where the `(` is actually found, which doesn't seem to match the comment. However, it does match the parser it gets passed to: https://github.com/fsharp/FsAutoComplete/blob/master/FSharp.CompilerBinding/Parser.fs#L302

The `($)` example I pointed out above preferably succeed, maybe just by searching from the column *before* the cursor rather than the one *at* the cursor. This would be as simple as changing the match below to pass it `col-1` instead. It might be better though to use the column after the paren and adjust the parser correspondingly.

I wonder if we could bail out at some point? If the file is very large and we are not in a method call this will go back to the start of the file. I thought of seeing a `=` when `depth` is 0, but that could be part of an expression argument of type `bool`. Perhaps a cap on the number of lines to go back? Seeing a `let`? Any thoughts?